### PR TITLE
Brian/core interfaces

### PR DESCRIPTION
**Pull request by bweick**
Added Interfaces for Vault and TransferProxy, Replaced full contracts in Core with Interfaces

### CODE REVIEW

File: contracts/core/interfaces/ITransferProxy.sol [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+*/
+
+pragma solidity 0.4.24;
+pragma experimental "ABIEncoderV2";
+
+/**
+ * @title ITransferProxy
+ * @author Set Protocol
+ *
+ * The IVault interface is used as a way to
```
Comment (justinkchen) [inline contracts/core/interfaces/ITransferProxy.sol]: Missing comment here

File: contracts/core/interfaces/IVault.sol [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+*/
+
+pragma solidity 0.4.24;
+pragma experimental "ABIEncoderV2";
+
+/**
+ * @title IVault
+ * @author Set Protocol
+ *
+ * The IVault interface is used as a way to
```
Comment (justinkchen) [inline contracts/core/interfaces/IVault.sol]: Missing the rest of this comment

File: contracts/core/interfaces/IVault.sol [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+     * @param  _tokenAddress    The address of the ERC20 token
+     * @param  _quantity        The number of tokens to deattribute to owner
+     */
+    function decrementTokenOwner(
+        address _owner,
+        address _tokenAddress,
+        uint _quantity
+    )
+        external;
+}
```
Comment (justinkchen) [inline contracts/core/interfaces/IVault.sol]: set your text editor to save with a newline at the end.  i dont know what you're using but if you're using something like Sublime, change your settings to have
"ensure_newline_at_eof_on_save": true
and
"trim_trailing_white_space_on_save": true

File: contracts/core/interfaces/ITransferProxy.sol [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.4.24;
+pragma experimental "ABIEncoderV2";
```
Comment (felix2feng) [inline contracts/core/interfaces/ITransferProxy.sol]: Are we getting rid of "ABIEncoderV2" for now?

**Pull request merged**